R/snasphot-external.R

```diff
@@ -0,0 +1,54 @@
+pandoc_path <- function() {
+  if (!is.na(path <- Sys.getenv("RSTUDIO_PANDOC", unset = NA))) {
+    return(file.path(path, "pandoc"))
+  }
+
+  if (nzchar(path <- Sys.which("pandoc"))) {
+    return(unname(path))
+  }
+
+  NULL
+}
+
+quarto_path <- function() {
+  if (!is.na(path <- Sys.getenv("QUARTO_PATH", unset = NA))) {
+    return(path)
+  }
+
+  if (nzchar(path <- Sys.which("quarto"))) {
+    return(unname(path))
+  }
+
+  NULL
+}
+
+pandoc_version <- function() {
+  path <- pandoc_path()
+  if (is.null(path)) {
+    return(NULL)
+  }
+
+  out <- system2(path, "--version", stdout = TRUE)
```

Comment:
Do we need this? https://github.com/rstudio/rmarkdown/blob/10863429024e236389a48f40ab1485620d53d311/R/pandoc.R#L720-L758

Reply:
Hmmmm, I'm guessing no given that the motivating issue was filed in 2014, and I've never heard of it.

Reply:
I remember this! Pandoc would consume massive amounts of memory (and then get OOM killed) if the "right" environment variables were not set. I think we are very careful about setting these variables in Connect and the IDE and avoid this problem.

Reply:
Ok, if you think it's a potential problem, I'll replicate the env vars.

Reply:
I definitely recall seeing this (I think mainly on Windows?) as well. I think we're better off keeping that code around just in case; it doesn't hurt.